### DISCUSSION TITLE
Using `Chem.CanonSmiles` in `stk.Smiles()` for better canonicalisation versus canonicalising the molecule.

### DISCUSSION THREAD

**Discussion by andrewtarzia** [lukasturcani/stk#537] [Q&A]
I have had an example where the canonicalisation in `stk.Smile().get_key(stk.Molecule)` does not find equivalent molecules when they are built from different topology graphs in `stk`. But if I use `Chem.CanonSmiles` on the smiles from those molecules through rdkit, they are found to be matching. 

Should we change algorithms? Or add a secondary algorithm? Or a warning?

Example of this implementation can be found in the new `stko` example: https://github.com/andrewtarzia/stko/blob/a57cec0ccac12670ad5b1a8209a783d23de64855/examples/intermediates_example.py

To implement the class: `stko.UnreactedTopologyGraph`, I needed to check for duplicate smiles in the `get_reacted_smiles` method, which would not work with the `stk` implementation of canonicalisation:

```python
    def get_reacted_smiles(self, n: int | None = None) -> set[str]:
        """Yield constructed molecules with n reactions performed."""
        yielded_smiles = set()
        for const_mol in self.yield_constructed_molecules(n=n):
            distinct_molecules = self.separate_molecule(const_mol)
            for dmol, _ in distinct_molecules:
                smiles = Chem.CanonSmiles(
                    Chem.MolToSmiles(dmol.to_rdkit_mol())
                )

                if "." in smiles:
                    msg = "Found `.` in smiles."
                    raise RuntimeError(msg)

                if smiles in yielded_smiles:
                    continue
                yielded_smiles.add(smiles)

        return yielded_smiles


```

**Comment by lukasturcani** [comment]
link example for posterity?
